code/espurna/utils.ino

```diff
@@ -6,31 +6,6 @@ Copyright (C) 2017-2018 by Xose Pérez <xose dot perez at gmail dot com>
 
 */
 
-// Core version 2.4.2 and higher changed the cont_t structure to a pointer:
-// https://github.com/esp8266/Arduino/commit/5d5ea92a4d004ab009d5f642629946a0cb8893dd#diff-3fa12668b289ccb95b7ab334833a4ba8L35
-#if defined(ARDUINO_ESP8266_RELEASE_2_3_0) \
-    || defined(ARDUINO_ESP8266_RELEASE_2_4_0) \
-    || defined(ARDUINO_ESP8266_RELEASE_2_4_1)
-extern "C" {
-    #include <cont.h>
-    extern cont_t g_cont;
-
-}
-
-unsigned int getFreeStack() {
-    return cont_get_free_stack(&g_cont);
-}
-#else
```

Comment:
can we add elif here just like in https://github.com/xoseperez/espurna/pull/1208/files#diff-52135e36de9de9ff791c1184a170b897R50, this way for 2.5.0 we won't have that extern "C" and rest.
Sorry if this is a bumb comment, I recently start learning Arduino and stuff.

Reply:
much less things on the screen was the main reason.
i am not an expert on c++, just felt like a better approach :) and as it turns out from #1205 all that multiline constructs inside .ino broke compilation when on Windows. moving to header also helped with that